### PR TITLE
feat: add coralogix dialect to opentelemetry

### PR DESCRIPTION
**Pull request by povilasv**
# Required for all PRs

- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

Added an Coralogix dialect to opentelemetry output plugin so that users can send metrics to Coralogix.

### CODE REVIEW

File: plugins/outputs/opentelemetry/README.md [hunk 1]
```diff
@@ -32,6 +32,18 @@ and agents via gRPC.
...
+  # dialect = "coralogix"
+
+  ## Your Coralogix private key is sensitive
+  # private_key: "xxx"
+
+  ## Metrics emitted by this plugin should be tagged
+  ## in Coralogix with the following application and subsystem names
+  # application_name: "$NAMESPACE"
+  # subsystem_name: "$HOSTNAME"
+
```
Comment (srebhan) [inline plugins/outputs/opentelemetry/README.md:46]: Maybe spend some more words in the config... How about adding some section like
```markdown
## Supported dialects

### Coralogix

This plugins supports sending data to a [Coralogix](https://coralogix.com) server by enabling the corresponding
dialect by uncommenting the `[output.telemetry.coralogix]` section. There, you can find the required setting to
interact with the server. The `private_key` can be found at <some-link-to-docu>.... Set the `application` and 
`subsystem` according to <how to determine the settings>...
```

**Review by srebhan — COMMENTED** [review]
Wow this is really nice @povilasv! Thanks for your effort! I've a suggestion regarding the configuration scheme and like to ask you to give users some more guidance by adding some more help to the README. What do you think?

File: plugins/outputs/opentelemetry/sample.conf [hunk 2]
```diff
@@ -24,6 +24,17 @@
...
+  ## Configuration options for the Coralogix dialect
+  ## Enable the following section of you use this plugin with a Coralogix endpoint
+  # [outputs.opentelemetry.coralogix]
+  #   ## Your Coralogix private key (required).
+  #   ## Please note that this is sensitive data!
+  #   private_key: "your_coralogix_key"
+  #
+  #   ## Application and subsystem names for the metrics (required)
+  #   application: "$NAMESPACE"
+  #   subsystem: "$HOSTNAME"
```
Comment (srebhan) [inline plugins/outputs/opentelemetry/sample.conf:36]: We are usually using `=` in TOML, so I suggest to keep it consistent...
```suggestion
  #   ## Your Coralogix private key (required).
  #   ## Please note that this is sensitive data!
  #   private_key = "your_coralogix_key"
  #
  #   ## Application and subsystem names for the metrics (required)
  #   application = "$NAMESPACE"
  #   subsystem = "$HOSTNAME"
```

File: plugins/outputs/opentelemetry/README.md [hunk 3]
```diff
@@ -41,6 +52,23 @@ and agents via gRPC.
...
+This plugins supports sending data to a [Coralogix](https://coralogix.com)
+server by enabling the corresponding dialect by uncommenting
+the `[output.opentelemetry.coralogix]` section.
+
+There, you can find the required setting to interact with the server.
+
+- The `private_key` is your Private Key, which you can find in Settings > Send Your Data.
+- The `application_name`, is your application name, which will be added to your metric attributes.
+- The `subsystem_name`, is your subsystem, which will be added to your metric attributes.
```
Comment (srebhan) [inline plugins/outputs/opentelemetry/README.md:67]:
```suggestion
- The `application`, is your application name, which will be added to your metric attributes.
- The `subsystem`, is your subsystem, which will be added to your metric attributes.
```

File: plugins/outputs/opentelemetry/opentelemetry.go [hunk 4]
```diff
@@ -44,6 +51,12 @@ type OpenTelemetry struct {
 	callOptions          []grpc.CallOption
 }
 
+type CoralogixConfig struct {
+	AppName    string `toml:"application_name"`
+	SubSystem  string `toml:"subsystem_name"`
```
Comment (srebhan) [inline plugins/outputs/opentelemetry/opentelemetry.go:56]:
```suggestion
	AppName    string `toml:"application"`
	SubSystem  string `toml:"subsystem"`
```

**Review by srebhan — COMMENTED** [review]
The code looks good @povilasv. Please use `=` in the TOML config and make sure that the `sample.conf` setting-names are reflected in the code. I think I marked all critical locations in the code, but please check again!

**Review by srebhan — APPROVED** [review]
Beautiful. Looks good to me. Thanks for your effort and willingness to help us keeping maintainability @povilasv!

**Review by powersj — APPROVED** [review]
Thanks for this!

**Pull request merged**